src/main/java/com/targetindia/DigitizeTimeTable/controller/StudentController.java

```diff
@@ -0,0 +1,105 @@
+//package com.targetindia.DigitizeTimeTable.controller;
+//
+//import com.targetindia.DigitizeTimeTable.service.AdminService;
+//import com.targetindia.DigitizeTimeTable.service.StudentService;
+//import org.springframework.beans.factory.annotation.Autowired;
+//import org.springframework.web.bind.annotation.GetMapping;
+//import org.springframework.web.bind.annotation.PathVariable;
+//import org.springframework.web.bind.annotation.RequestMapping;
+//import org.springframework.web.bind.annotation.RestController;
+//
+//import java.util.ArrayList;
+//import java.util.HashMap;
+//
+//@RestController
+//@RequestMapping("student")
+//public class StudentController {
+//    @Autowired
+//    StudentService service;
+//
+//    @GetMapping("/GET/time_table/class_id/{class_id}/day/{day}")
+//    public HashMap<String, ArrayList<String>> getStudentDayTimeTable(@PathVariable int class_id, @PathVariable String day){
+//        return service.getStudentDayTimeTable(class_id, day);
+//    }
+//
+//    @GetMapping("/time_table/{class_id}")
+//    public HashMap<String, HashMap<String, ArrayList<String>>> getStudentWeeklyTimeTable(@PathVariable int class_id){
+//        return service.getStudentWeeklyTimeTable(class_id);
+//    }
+//}
+package com.targetindia.DigitizeTimeTable.controller;
+
+import com.targetindia.DigitizeTimeTable.service.AdminService;
+import com.targetindia.DigitizeTimeTable.service.StudentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Optional;
+
+import static java.time.Year.isLeap;
+
+@RestController
+@RequestMapping("students")
+public class StudentController {
+    @Autowired
+    StudentService service;
+
+    @GetMapping("/time_table/{class_id}/{date}")
+    public ResponseEntity<String> getStudentDayTimeTable(@PathVariable int class_id, @PathVariable String date){
+        String[] date_array = date.split("-",3);
+
+
+        int day = Integer.parseInt(date_array[0]);
```

Comment:
Move the date validation logic (getDaysOfMonth, isLeap) to a separate utility class, because this is not really the responsibility of a controller.